src/utils/mediaDevicesUtils.js

```diff
@@ -0,0 +1,154 @@
+import store from '@/plugins/vuex';
+import { stopIndicesWithElidedDims } from '@tensorflow/tfjs-core/dist/ops/slice_util';
+
+navigator.mediaDevices.ondevicechange = async function() {
+    refreshMediaDevices();
+    await updateCurrentStream();
+}
+
+async function updateCurrentStream() {
+    const userControl = store.getters.userControl;
+
+    if (userControl) {
+        console.log("updating publish stream")
+        await republishVideo();
+        await republishAudio();
+        return;
+    }
+
+    console.log("updating local stream")
+    const localStream = await getLocalStream();
+    store.commit('setLocalStream', localStream);
+}
+
+async function getLocalStream() {
+    const tracks = [];
+
+    const updatedAudioStream = await updateAudioStream();
+    const updatedVideoStream = await updateVideoStream();
+
+    tracks.push(updatedAudioStream);
+    tracks.push(updatedVideoStream);
+
+    const activeTracks = tracks.filter(
+        track => track && track.readyState === 'live'
+    );
+
+    if (activeTracks.length <= 0) {
+        return null;
+    }
+
+    return new MediaStream(activeTracks);
+}
+
+async function republishVideo() {
+    const userControl = store.getters.userControl;
+
+    console.log("republishing video")
+    if (!store.getters.videoActive && store.getters.localUser.stream.getVideoTracks().length > 0) {
+        userControl.stopVideoTrack();
```

Comment:
function says republish but in this flow video is not published, plz change name to reflect what the function does